### ISSUE TITLE
JointLimits grow over time

### ISSUE THREAD

**Issue by mpowelson** [ros-industrial-consortium/tesseract#445]
@Levi-Armstrong Using our new change joint limits code I am getting errors like this. It seems every time we call it, the limits matrix in the forward kinematics object grows by n_dof

```
[ WARN] ros.rosconsole_bridge.console_bridge: Invalid limits when clamping Waypoint. Waypoint size: 6, Limits size: 30
[ WARN] ros.rosconsole_bridge.console_bridge: Invalid limits when clamping Waypoint. Waypoint size: 6, Limits size: 30
```

**Comment by mpowelson** [comment]
KDL setLimits looks fine to me. I'm thinking it must be upstream of there.

**Comment by Levi-Armstrong** [comment]
Are you applying it using the command?

**Comment by mpowelson** [comment]
I'm not doing it via ROS. I'm using `tesseract->getEnvironment()->changeJointPositionLimits(const std::string& joint_name, double lower, double upper)`

**Comment by Levi-Armstrong** [comment]
I see the issue. In the kinematics init method it should be clearing out the internal data member.. So when we call init in the update the data is getting appended.

**Comment by mpowelson** [comment]
Ok. Well I just wrote a unit test that fails. I'll change that and make sure it passes now.

**Comment by mpowelson** [comment]
Do you want me to just do kdl_data_ = KDLChainData() or should we add a clear method?

**Comment by Levi-Armstrong** [comment]
I think that should do it. Everything else looks to get overridden.  I am also good with the clear method.

**Comment by mpowelson** [comment]
Ok. That did seem to fix it. I'll probably just leave it to that for now.

**Comment by Levi-Armstrong** [comment]
Addressed by #446
